frame/evm-system/src/lib.rs

```diff
@@ -160,16 +160,21 @@ impl<T: Config> Pallet<T> {
 
 	/// Increment a particular account's nonce by 1.
 	pub fn inc_account_nonce(who: &<T as Config>::AccountId) {
-		Account::<T>::mutate(who, |a| {
-			a.nonce += <T as Config>::Index::one();
-
-			// Meaning that account is being created.
-			if a.nonce == <T as Config>::Index::one()
-				&& a.data == <T as Config>::AccountData::default()
-			{
-				Self::on_created_account(who.clone());
-			}
+		let is_new_account = Account::<T>::mutate_exists(who, |maybe_account| {
+			let is_new_account = maybe_account.is_none();
+
+			let mut account = maybe_account.take().unwrap_or_default();
+			account.nonce += <T as Config>::Index::one();
+
+			*maybe_account = Some(account);
+
+			is_new_account
 		});
```

Comment:
```suggestion
		let is_new_account = Account::<T>::mutate_exists(who, |maybe_account| {
			let is_new_account = maybe_account.is_none();

			let mut account = maybe_account.get_or_insert_default();
			account.nonce += <T as Config>::Index::one();

			is_new_account
		});
```

Reply:
Thanks, haven't used it before, but it looks interesting!